common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java

```diff
@@ -98,267 +98,360 @@
+                if (allowItemPipeBlock(material, itemPipeType)) {
+                    registerItemPipeBlock(material, itemPipeType);
                 }
             }
         }
-        ITEM_PIPE_BLOCKS = itemsBuilder.build();
+        GTCEu.LOGGER.info("Generating Item Pipe Blocks... Complete!");
+    }
+    public static boolean allowItemPipeBlock(Material material, ItemPipeType itemPipeType) {
+        return material.hasProperty(PropertyKey.ITEM_PIPE)
+                && !itemPipeType.getTagPrefix().isIgnored(material);
+    }
+    public static void registerItemPipeBlock(Material material, ItemPipeType itemPipeType) {
+        var entry = REGISTRATE.block("%s_%s_item_pipe".formatted(material.getName(), itemPipeType.name), p -> new ItemPipeBlock(p, itemPipeType, material))
+                .initialProperties(() -> Blocks.IRON_BLOCK)
+                .properties(p -> {
+                    if (doMetalPipe(material)) {
+                        p.sound(GTSoundTypes.METAL_PIPE);
+                    }
+                    return p.dynamicShape().noOcclusion().noLootTable();
+                })
+                .transform(unificationBlock(itemPipeType.getTagPrefix(), material))
+                .blockstate(NonNullBiConsumer.noop())
+                .setData(ProviderType.LANG, NonNullBiConsumer.noop())
+                .setData(ProviderType.LOOT, NonNullBiConsumer.noop())
+                .addLayer(() -> RenderType::cutoutMipped)
+                .color(() -> MaterialPipeBlock::tintedColor)
+                .item(MaterialPipeBlockItem::new)
+                .model(NonNullBiConsumer.noop())
+                .color(() -> MaterialPipeBlockItem::tintColor)
+                .build()
+                .register();
+        ITEM_PIPE_BLOCKS_BUILDER.put(itemPipeType.getTagPrefix(), material, entry);
     }
 
+    // Laser Pipe Blocks
+    public static void generateLaserPipeBlocks() {
+        for (int i = 0; i < LaserPipeType.values().length; ++i) {
+            registerLaserPipeBlock(i);
+        }
+    }
+    public static void registerLaserPipeBlock(int slot) {
+        var type = LaserPipeType.values()[slot];
+        var entry = REGISTRATE.block("%s_laser_pipe".formatted(type.getSerializedName()), (p) -> new LaserPipeBlock(p, type))
+                .initialProperties(() -> Blocks.IRON_BLOCK)
+                .properties(p -> p.dynamicShape().noOcclusion().noLootTable())
+                .blockstate(NonNullBiConsumer.noop())
+                .defaultLoot()
+                .tag(GTToolType.WIRE_CUTTER.harvestTag)
+                .addLayer(() -> RenderType::cutoutMipped)
+                .color(() -> LaserPipeBlock::tintedColor)
+                .item(LaserPipeBlockItem::new)
+                .model(NonNullBiConsumer.noop())
+                .color(() -> LaserPipeBlockItem::tintColor)
+                .build()
+                .register();
+        LASER_PIPES[slot] = entry;
```

Comment:
Might be better to return the entry here and assign it in the calling function, could possibly even use something like this if you change the param type to `LaserPipeType`:

```java
LASER_PIPES = Arrays.stream(LaserPipeType.values())
    .map(GTBlocks::registerLaserPipeBlock)
    .toArray(BlockEntry[]::new)
```